### PR TITLE
Skip default spreading scoring plugin for pods that define TopologySpreadConstraints

### LINKED ISSUE #87521: Skip default spreading scoring plugin for pods that define TopologySpreadConstraints
Currently the scheduler by default spreads the pods of a deployment based on a hard-coded [criteria](https://github.com/kubernetes/kubernetes/blob/90da466221c14e76b4509ddc310e7a7bd4e0039c/pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go#L73). 

We recently added a pod object api (pod.Spec.TopologySpreadConstraints) that allows specifying spread behavior at the workload level. For pods that specify topology spread constraints, the default spreading behavior should not apply.

/sig scheduling
/cc @Huang-Wei @alculquicondor

### PR DESCRIPTION
**Pull request by Ye-Tian-Zero**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
For pods that define TopologySpreadConstraints, we are using a workload level spread scoring method for it, so we should skip the default spread scoring for these pods.

**Which issue(s) this PR fixes**:

Fixes #87521 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Skip default spreading scoring plugin for pods that define TopologySpreadConstraints
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go [hunk 1]
```diff
@@ -67,10 +67,20 @@ func (s *postFilterState) Clone() framework.StateData {
 	return s
 }
 
+// skipDefaultPodTopologySpread returns true if the pod's TopologySpreadConstraints is specified
+// The pod's spread behavior will be calculated in workload level
+func skipDefaultPodTopologySpread(pod *v1.Pod) bool {
+	return pod.Spec.TopologySpreadConstraints != nil
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go]: Prefer

```suggestion
	return len(pod.Spec.TopologySpreadConstraints) != 0
```

**Review by Huang-Wei — COMMENTED** [review]
A nit below.

File: pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go (hunk 1 above)
Comment (skilxn-go) [reply pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go]: Thanks, fixed

File: pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go [hunk 2]
```diff
@@ -67,10 +67,20 @@ func (s *postFilterState) Clone() framework.StateData {
 	return s
 }
 
+// skipDefaultPodTopologySpread returns true if the pod's TopologySpreadConstraints is specified
```
[the remark this answers is not in the document]
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go]: +1, and please end the sentence with a period.

File: pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go [hunk 3]
```diff
@@ -67,10 +67,20 @@ func (s *postFilterState) Clone() framework.StateData {
...
+// The pod's spread behavior will be calculated in workload level
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go]: line seems redundant, please remove it.

**Pull request merged**